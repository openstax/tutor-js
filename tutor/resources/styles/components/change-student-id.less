.change-id-panel {
  max-width: 840px;
  padding: 30px 80px;

  .alert-danger {
    color: @openstax-danger;
    background: @openstax-info-bg;
  }
  .panels {
    width: 80%;


    height: 110px;

    input[type="text"] {
      height: 45px;
      width: 95%;
    }

    button {
      .btn-tutor-primary();
    }

  }

  .cancel {
    margin-top: 40px;
  }
}
