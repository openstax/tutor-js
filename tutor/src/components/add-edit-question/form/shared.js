import { React, PropTypes, styled, cn, observer, useState } from 'vendor';
import { Form, Overlay, Popover } from 'react-bootstrap';
import { colors } from 'theme';
import { Icon } from 'shared';
import { EditableHTML } from '../../editor';

const StyledRowContent = styled.div`
  display: flex;
  flex-wrap: wrap;
  >.label-wrapper {
    flex: 0 1 10%;
    font-size: 1.4rem;
    line-height: 3.2rem;
    font-weight: 700;
    color: ${colors.neutral.darker};
    text-transform: uppercase;
    &.gray-background {
      padding-top: 1.6rem;
    }
  }
  >.content-form {
    flex: 0 1 90%;
    flex-wrap: wrap;
    &.gray-background {
      background-color: ${colors.neutral.bright};
    }
    &.padding {
      padding: 2.4rem 1.8rem;
    }
  }
  .editor-wrapper {
    .limited {
      .editable-html {
        min-height: 80px;
      }
    }
    .editor {
      .error-info {
        color: ${colors.strong_red};
      }
      .editable-html {
        margin: 0;
        background-color: white;
        border: 1px solid ${colors.neutral.pale};
        .openstax-has-html {
          margin: 1rem;
          // placeholder are not in a <p> tag
          color: ${colors.neutral.thin};
          p {
            color: ${colors.neutral.darker};
            line-height: 2.4rem;
            font-size: 1.6rem;
          }
        }
      } 
      // overriding the inline style
      .perry-white {
        border: 1px solid ${colors.neutral.pale};
        width: 100% !important;
      }
    }
    &.isEmpty .editable-html {
      border: 1px solid ${colors.soft_red};
      background-color: ${colors.gray_red};
    }
  }
`;

const StyledAddEditFormTextInput = styled(Form.Group)`
    input {
        color: ${colors.neutral.darker};
        &::placeholder {
            font-size: 1.4rem;
            color: ${colors.neutral.thin};
        }
        &:focus {
            background-color: ${colors.white};
        }
    }
`;

const StyledPopover = styled(Popover)`
  padding: 1.5rem;
  font-size: 1.4rem;
  p {
    color: ${colors.neutral.darker};
  }
  a {
    font-weight: 500;
  }
`;

const StyledQuestionInfoIcon = styled(Icon)`
  &.question-info-icon {
    margin-left: 1rem;
    color:${colors.bright_blue};
    margin: 0;
  }
`;

export const QuestionInfo = ({ popoverInfo, placement = 'top' }) => {
  const [show, setShow] = useState(false);
  const [target, setTarget] = useState(null);

  const handleClick = (event) => {
    setShow(!show);
    if(!show) {
      setTarget(event.target);
    }
    else {
      setTarget(null);
    }
  };
  const popover = <StyledPopover>
    {popoverInfo}
  </StyledPopover>;

  return (
    <>
      <StyledQuestionInfoIcon
        type="question-circle"
        className="question-info-icon"
        onClick={handleClick}
      />
      <Overlay
        rootClose
        show={show}
        target={target}
        placement={placement}
        onHide={() => setShow(false)}
      >
        {popover}
      </Overlay>
    </>
  
  );
};
QuestionInfo.propTypes = {
  popoverInfo: PropTypes.node.isRequired,
  placement: PropTypes.string,
};


const EditableHTMLPanel = styled(EditableHTML)({
  flex: 1,
  '.ProseMirror.pw-prosemirror-editor': {
    padding: '2rem',
  },
});


export const TextInputHTMLEditor = ({ className, label, errorInfo, html, ...props }) => (
  <StyledRowContent className={cn('editor-wrapper', className, { 'isEmpty': errorInfo })}>
    {label && <Form.Label>{label}</Form.Label>}
    <div className="editor limited">
      <EditableHTMLPanel {...props} limitedEditing={true} html={html || ''}/>
      {errorInfo && <p className="error-info">{errorInfo}</p>}
    </div>
  </StyledRowContent>
);
TextInputHTMLEditor.propTypes = {
  html: PropTypes.string,
  label: PropTypes.string,
  className: PropTypes.string,
  errorInfo: PropTypes.string,
};

export const AnswerHTMLEditor = ({ className, label, errorInfo, html, ...props }) => (
  <StyledRowContent className={cn('editor-wrapper', className, { 'isEmpty': errorInfo })}>
    {label && <Form.Label>{label}</Form.Label>}
    <div className="editor">
      <EditableHTMLPanel {...props} limitedEditing={false} html={html || ''}/>
      {errorInfo && <p className="error-info">{errorInfo}</p>}
    </div>
  </StyledRowContent>
);
AnswerHTMLEditor.propTypes = {
  html: PropTypes.string,
  label: PropTypes.string,
  className: PropTypes.string,
  errorInfo: PropTypes.string,
};

export const AddEditFormTextInput = observer(({ onChange, plainText, value, label, placeholder, className, errorInfo }) => {
  const input = plainText ?
    <Form.Control type="text" onChange={({ target: { value } }) => onChange(value)} value={value} placeholder={placeholder} /> :
    <TextInputHTMLEditor onChange={onChange} html={value} placeholder={placeholder} errorInfo={errorInfo} />;
  return (
    <StyledAddEditFormTextInput controlId={className} className={className}>
      {label && <Form.Label>{label}</Form.Label>}
      {input}
    </StyledAddEditFormTextInput>
  );
});

export const AddEditQuestionFormBlock = ({
  label,
  formContentRenderer: FormContent,
  className,
  showGrayBackground = false,
  addPadding = true,
  onFocus,
}) => {
  return (
    <StyledRowContent className={className} onFocus={onFocus ? onFocus : null}>
      <div
        className={cn('label-wrapper', { 'gray-background': showGrayBackground })}>
        <span>{label}</span>
      </div>
      <div
        className={cn('content-form', { 'gray-background': showGrayBackground, 'padding': addPadding })}>
        <FormContent />
      </div>
    </StyledRowContent>
  );
};
AddEditQuestionFormBlock.propTypes = {
  label: PropTypes.string.isRequired,
  formContentRenderer: PropTypes.func.isRequired,
  className: PropTypes.string,
  showGrayBackground: PropTypes.bool,
  addPadding: PropTypes.bool,
  onFocus: PropTypes.func,
};
