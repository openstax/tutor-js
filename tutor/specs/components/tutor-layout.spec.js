import { Router as R } from '../helpers';
import { TutorLayout } from '../../src/components/tutor-layout';
import MM from '../../src/components/modal-manager';
import TC from '../../src/components/tours/conductor';
import { currentUser as User } from '../../src/models';
import { SpyMode as SM } from 'shared';

describe('Tutor Layout', () => {
    User.bootstrap({});

    it('renders and matches snapshot', () => {
        const l = mount(
            <R>
                <SM.Wrapper>
                    <MM>
                        <TC>
                            <TutorLayout>
                                <p>hi i am body</p>
                            </TutorLayout>
                        </TC>
                    </MM>
                </SM.Wrapper>
            </R>
        );
        expect(l.debug()).toMatchSnapshot();
        l.unmount();
    });

});
