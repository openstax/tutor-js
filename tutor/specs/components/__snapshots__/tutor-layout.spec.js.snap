// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutor Layout renders and matches snapshot 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <SpyModeWrapper>
      <MobXProvider spyMode={{...}}>
        <div className=\\"openstax-debug-content\\">
          <ModalManager>
            <MobXProvider modalManager={{...}}>
              <inject-with-modalManager-spyMode(TourConductor)>
                <TourConductor modalManager={{...}} spyMode={{...}}>
                  <MobXProvider tourContext={{...}}>
                    <div data-purpose=\\"tour-conductor-wrapper\\">
                      <TutorLayout>
                        <MobXProvider topNavbar={{...}} courseContext={{...}} bottomNavbar={{...}} setSecondaryTopControls={[Function: res]}>
                          <styled.div hasNavbar={true}>
                            <div className=\\"sc-fznKkj fZlCeP\\">
                              <Navbar area=\\"header\\" context={{...}} isDocked={false}>
                                <styled.nav area=\\"header\\" shouldHideNavbar={[undefined]} isDocked={false} className=\\"tutor-navbar\\">
                                  <nav className=\\"sc-AxjAm fhzKny tutor-navbar\\">
                                    <div className=\\"left-side-controls\\">
                                      <Component>
                                        <Logo>
                                          <Memo(wrappedComponent) courseContext={{...}}>
                                            <TutorLink to=\\"myCourses\\" className=\\"brand\\" aria-label=\\"dashboard\\">
                                              <Link to={{...}} className=\\"brand\\" aria-label=\\"dashboard\\">
                                                <LinkAnchor className=\\"brand\\" aria-label=\\"dashboard\\" href=\\"/dashboard\\" navigate={[Function: navigate]}>
                                                  <a className=\\"brand\\" aria-label=\\"dashboard\\" href=\\"/dashboard\\" onClick={[Function: onClick]}>
                                                    <styled.svg viewBox=\\"0 0 264.62 40.7\\">
                                                      <svg viewBox=\\"0 0 264.62 40.7\\" className=\\"sc-fzpans iRdary\\">
                                                        <defs>
                                                          <style />
                                                        </defs>
                                                        <g id=\\"prefix__Layer_2\\" data-name=\\"Layer 2\\">
                                                          <g id=\\"prefix__Layer_1-2\\" data-name=\\"Layer 1\\">
                                                            <path d=\\"M64.85 9a2.12 2.12 0 0 1-2.22 2L13.16 8.63a2.12 2.12 0 0 1-2-2.22L11.35 2a2.12 2.12 0 0 1 2.22-2L63 2.35a2.12 2.12 0 0 1 2 2.22z\\" fill=\\"#78b042\\" />
                                                            <rect fill=\\"#5f6062\\" x={5.81} y={17.47} width={42.78} height={8.83} rx={1.27} ry={1.27} />
                                                            <path d=\\"M61 31.83a1.25 1.25 0 0 1-1.19 1.31l-45.4 1.42a1.25 1.25 0 0 1-1.27-1.23L13 29.59a1.25 1.25 0 0 1 1.19-1.31l45.4-1.42a1.25 1.25 0 0 1 1.27 1.23z\\" fill=\\"#f4cf1c\\" />
                                                            <path d=\\"M54.87 39.43a1.23 1.23 0 0 1-1.2 1.27H10.34a1.23 1.23 0 0 1-1.2-1.27v-2.5a1.24 1.24 0 0 1 1.2-1.27h43.33a1.24 1.24 0 0 1 1.2 1.27z\\" fill=\\"#242f66\\" />
                                                            <path fill=\\"#f47641\\" d=\\"M56 14.45c0 .52-.77.94-1.71.94H1.71C.77 15.38 0 15 0 14.45v-2c0-.52.77-.94 1.71-.94h52.56c.95 0 1.71.42 1.71.94z\\" />
                                                            <path fill=\\"#5f6062\\" d=\\"M80.41 16.36A5.35 5.35 0 0 1 86 21.91a5.55 5.55 0 1 1-11.09 0 5.35 5.35 0 0 1 5.5-5.55zm0 9.86a4.09 4.09 0 0 0 4.18-4.31 4.18 4.18 0 1 0-8.35 0 4.09 4.09 0 0 0 4.17 4.31zM88.72 16.64h1.37v1.85a4.86 4.86 0 0 1 4.22-2.12 5.35 5.35 0 0 1 5.55 5.54 5.35 5.35 0 0 1-5.55 5.55 4.86 4.86 0 0 1-4.22-2.12v7.05h-1.37zm5.64 9.58a4.09 4.09 0 0 0 4.18-4.31 4.09 4.09 0 0 0-4.18-4.31 4.31 4.31 0 1 0 0 8.63zM112.27 25.2a5.18 5.18 0 0 1-4.56 2.26c-3.22 0-5.34-2.3-5.34-5.57a5.24 5.24 0 0 1 5.27-5.52 4.79 4.79 0 0 1 5 4.93v1h-8.86a3.93 3.93 0 0 0 4 4 4.24 4.24 0 0 0 3.54-1.87zm-1-4.18a3.55 3.55 0 0 0-3.65-3.42 3.91 3.91 0 0 0-3.88 3.4zM115.65 19.6c0-.78-.12-2.42-.12-3h1.37c0 .77 0 1.64.09 1.85h.07a3.94 3.94 0 0 1 3.61-2.12c3.29 0 4 2.24 4 4.54v6.28h-1.37V21c0-1.71-.48-3.38-2.65-3.38S117 19 117 21.91v5.27h-1.37zM133.91 19.26a2.49 2.49 0 0 0-2.15-1.12c-.73 0-1.6.34-1.6 1.16 0 2 5.91.37 5.91 4.63 0 2.6-2.49 3.51-4.72 3.51a5.36 5.36 0 0 1-4.27-1.69l1.83-1.71a3.36 3.36 0 0 0 2.6 1.35c.8 0 1.83-.39 1.83-1.26 0-2.26-5.91-.48-5.91-4.66 0-2.44 2.19-3.54 4.4-3.54a4.74 4.74 0 0 1 3.9 1.67zM136.92 18.55v-2.32h2.26v-3.18h2.74v3.17h3v2.33h-3v4.79c0 1.1.32 1.78 1.53 1.78a2.84 2.84 0 0 0 1.48-.34v2.31a5.78 5.78 0 0 1-2.17.37c-2.9 0-3.58-1.3-3.58-3.88v-5zM153.93 25.68h-.07a3.94 3.94 0 0 1-3.51 1.78c-2 0-4-1.12-4-3.29 0-3.56 4.15-3.81 6.89-3.81h.69v-.3c0-1.35-1.05-2.05-2.51-2.05a4.31 4.31 0 0 0-2.92 1.14l-1.44-1.44a6.42 6.42 0 0 1 4.65-1.71c4.68 0 4.68 3.38 4.68 4.93v6.3h-2.46zm-.16-3.4h-.57c-1.51 0-4.11.11-4.11 1.69 0 1 1 1.44 1.92 1.44 1.87 0 2.76-1 2.76-2.51zM162.17 21.29l-3.79-5.07h3.33l2.19 3.29 2.37-3.29h3.13l-3.74 5.07 4.4 5.89h-3.33l-2.85-3.93-2.88 3.93h-3.24z\\" />
                                                            <path fill=\\"#f47641\\" d=\\"M186.42 17.41h-3.62v9.73h-1.3v-9.73h-3.61v-1.2h8.53zM196.7 23a6.31 6.31 0 0 1-.22 1.7 3.94 3.94 0 0 1-.71 1.41 3.44 3.44 0 0 1-1.27 1 5.24 5.24 0 0 1-3.83 0 3.43 3.43 0 0 1-1.27-1 3.94 3.94 0 0 1-.71-1.41 6.31 6.31 0 0 1-.22-1.7v-6.79h1.3v6.58a5.85 5.85 0 0 0 .14 1.28 3.18 3.18 0 0 0 .46 1.1 2.34 2.34 0 0 0 .86.76 3.38 3.38 0 0 0 2.72 0 2.35 2.35 0 0 0 .86-.76 3.19 3.19 0 0 0 .46-1.1 5.85 5.85 0 0 0 .14-1.28v-6.58h1.3zM207.27 17.41h-3.61v9.73h-1.3v-9.73h-3.61v-1.2h8.53zM214.43 27.42a5.92 5.92 0 0 1-2.29-.43 5.5 5.5 0 0 1-1.81-1.2 5.39 5.39 0 0 1-1.19-1.79 6.41 6.41 0 0 1 0-4.59 5.39 5.39 0 0 1 1.19-1.82 5.52 5.52 0 0 1 1.81-1.2 6.3 6.3 0 0 1 4.59 0 5.52 5.52 0 0 1 1.81 1.2 5.39 5.39 0 0 1 1.19 1.82 6.41 6.41 0 0 1 0 4.59 5.39 5.39 0 0 1-1.19 1.82 5.5 5.5 0 0 1-1.81 1.2 5.92 5.92 0 0 1-2.3.4zm0-1.2a4.25 4.25 0 0 0 1.75-.35 4.14 4.14 0 0 0 1.37-1 4.46 4.46 0 0 0 .89-1.44 5.1 5.1 0 0 0 0-3.54 4.45 4.45 0 0 0-.89-1.44 4.14 4.14 0 0 0-1.37-1 4.51 4.51 0 0 0-3.51 0 4.14 4.14 0 0 0-1.37 1 4.45 4.45 0 0 0-.89 1.44 5.1 5.1 0 0 0 0 3.54 4.46 4.46 0 0 0 .89 1.44 4.14 4.14 0 0 0 1.37 1 4.25 4.25 0 0 0 1.76.35zM222.79 16.21h2.8a13.33 13.33 0 0 1 1.57.09 4.21 4.21 0 0 1 1.4.4 2.54 2.54 0 0 1 1 .9 2.94 2.94 0 0 1 .39 1.61 2.66 2.66 0 0 1-.73 1.95 3.46 3.46 0 0 1-2 .94l3.1 5h-1.59l-3-4.91h-1.67v4.91h-1.3zm1.3 4.82H226.37a3.83 3.83 0 0 0 1.07-.2 1.85 1.85 0 0 0 .8-.53 1.54 1.54 0 0 0 .32-1 1.7 1.7 0 0 0-.23-.94 1.56 1.56 0 0 0-.61-.55 2.58 2.58 0 0 0-.85-.25 7.22 7.22 0 0 0-.95-.06h-1.82zM259.72 16.9h-.87v-.44H261v.44h-.87v2.32h-.44zm1.87-.44h.7l.82 2.1.83-2.1h.68v2.76h-.44V16.9l-.91 2.32H263l-1-2.32v2.32h-.44zM236.09 16.12h1l-.56 4a2.61 2.61 0 0 1 2-.91 2.27 2.27 0 0 1 2.38 2.79 3.15 3.15 0 0 1-3.13 2.82 1.94 1.94 0 0 1-1.73-.91l-.11.77h-1zm3.73 5.88a1.47 1.47 0 0 0-1.5-1.87 2.05 2.05 0 0 0-2 1.87 1.48 1.48 0 0 0 1.51 1.87 2 2 0 0 0 1.99-1.87zM242.92 22.38c-.14.94.65 1.56 1.59 1.56a2.17 2.17 0 0 0 1.59-.8l.69.59a3.26 3.26 0 0 1-2.54 1.1 2.28 2.28 0 0 1-2.38-2.83 3.23 3.23 0 0 1 3.13-2.82c1.89 0 2.41 1.45 2.22 2.83v.36zm3.33-.82a1.25 1.25 0 0 0-1.34-1.56 2 2 0 0 0-1.91 1.56zM251.63 20.2h-1.46l-.35 2.44c-.08.6-.17 1.24.6 1.24a1.57 1.57 0 0 0 .75-.16l-.13.93a2.56 2.56 0 0 1-1 .18c-1.58 0-1.46-1-1.34-1.82l.4-2.8H248l.12-.88h1.18l.2-1.51h1l-.2 1.51h1.46zM253 20a3.62 3.62 0 0 1 2.24-.79c1.59 0 2.13.86 2 1.79l-.38 2.74a8.46 8.46 0 0 0-.09 1h-.91c0-.27 0-.54.08-.82a2.38 2.38 0 0 1-2 1c-1 0-1.78-.57-1.63-1.61.19-1.38 1.59-1.86 3.22-1.86h.75v-.23c.08-.56-.25-1.13-1.12-1.13a2.53 2.53 0 0 0-1.61.61zm2.58 2.13c-1 0-2.16.17-2.29 1-.08.61.33.87 1 .87a1.72 1.72 0 0 0 1.76-1.56v-.34z\\" />
                                                          </g>
                                                        </g>
                                                      </svg>
                                                    </styled.svg>
                                                  </a>
                                                </LinkAnchor>
                                              </Link>
                                            </TutorLink>
                                          </Memo(wrappedComponent)>
                                        </Logo>
                                      </Component>
                                    </div>
                                    <div className=\\"center-control\\" />
                                    <div className=\\"right-side-controls\\">
                                      <Component>
                                        <Menus>
                                          <Memo(wrappedComponent) courseContext={{...}}>
                                            <Responsive desktop={{...}} mobile={{...}}>
                                              <Memo(wrappedComponent) course={[undefined]}>
                                                <StudentPayNowBtn course={[undefined]} />
                                                <ActionsMenu course={[undefined]} />
                                                <withRouter(inject(SupportMenu)) course={[undefined]}>
                                                  <inject(SupportMenu) course={[undefined]} history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                                                    <Observer>
                                                      <SupportMenu course={[undefined]} history={{...}} location={{...}} match={{...}} staticContext={[undefined]} tourContext={{...}}>
                                                        <Responsive desktop={{...}} mobile={{...}}>
                                                          <Dropdown show={true} onToggle={[Function: res]} navbar={false}>
                                                            <ReactOverlaysDropdown drop={[undefined]} show={true} alignEnd={[undefined]} onToggle={[Function]} focusFirstItemOnShow={[undefined]} itemSelector=\\".dropdown-item:not(.disabled):not(:disabled)\\">
                                                              <div onKeyDown={[Function: handleKeyDown]} className=\\"show dropdown\\">
                                                                <DropdownToggle id=\\"support-menu\\" aria-label=\\"Page tips and support\\" variant=\\"ox\\">
                                                                  <Button onClick={[Function]} className=\\"dropdown-toggle\\" aria-haspopup={true} aria-expanded={true} id=\\"support-menu\\" aria-label=\\"Page tips and support\\" variant=\\"ox\\" active={false} disabled={false} type=\\"button\\">
                                                                    <button onClick={[Function]} aria-haspopup={true} aria-expanded={true} id=\\"support-menu\\" aria-label=\\"Page tips and support\\" disabled={false} type=\\"button\\" className=\\"dropdown-toggle btn btn-ox\\">
                                                                      <inject(TourAnchor) id=\\"support-menu-button\\" aria-labelledby=\\"support-menu\\" onSelect={[Function: res]}>
                                                                        <Observer>
                                                                          <TourAnchor id=\\"support-menu-button\\" aria-labelledby=\\"support-menu\\" onSelect={[Function: res]} tourContext={{...}} tag=\\"div\\" className=\\"\\">
                                                                            <div className=\\"tour-anchor\\" data-tour-anchor-id=\\"support-menu-button\\" aria-labelledby=\\"support-menu\\" onSelect={[Function: res]}>
                                                                              <Icon type=\\"question-circle\\" buttonProps={{...}} tooltipProps={{...}}>
                                                                                <IconWrapper data-variant={[undefined]} className=\\"ox-icon ox-icon-question-circle\\" icon={{...}}>
                                                                                  <FontAwesomeIcon data-variant={[undefined]} className=\\"IconWrapper-ra27cp-0 gKKqUB ox-icon ox-icon-question-circle\\" icon={{...}} border={false} mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}}>
                                                                                    <svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"question-circle\\" className=\\"svg-inline--fa fa-question-circle fa-w-16 IconWrapper-ra27cp-0 gKKqUB ox-icon ox-icon-question-circle\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" style={{...}} data-variant={[undefined]}>
                                                                                      <path fill=\\"currentColor\\" d=\\"M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z\\" style={{...}} />
                                                                                    </svg>
                                                                                  </FontAwesomeIcon>
                                                                                </IconWrapper>
                                                                              </Icon>
                                                                              <span title=\\"Page tips and support\\" className=\\"control-label\\">
                                                                                Help
                                                                              </span>
                                                                            </div>
                                                                          </TourAnchor>
                                                                        </Observer>
                                                                      </inject(TourAnchor)>
                                                                    </button>
                                                                  </Button>
                                                                </DropdownToggle>
                                                                <DropdownMenu className=\\" hide\\" alignRight={false} flip={true}>
                                                                  <div x-placement=\\"bottom-start\\" aria-labelledby=\\"support-menu\\" style={{...}} className=\\" hide dropdown-menu show\\">
                                                                    <Memo(wrappedComponent) onPlayClick={[Function: res]} course={[undefined]} history={{...}} location={{...}} match={{...}} staticContext={[undefined]} tourContext={{...}} />
                                                                    <DropdownItem className=\\"-help-link\\" target=\\"_blank\\" onSelect={[Function: res]} href=\\"https://openstax.secure.force.com/help?search=tutor\\" as={{...}} disabled={false}>
                                                                      <SafeAnchor target=\\"_blank\\" href=\\"https://openstax.secure.force.com/help?search=tutor\\" disabled={false} className=\\"-help-link dropdown-item\\" onClick={[Function]}>
                                                                        <a target=\\"_blank\\" href=\\"https://openstax.secure.force.com/help?search=tutor\\" className=\\"-help-link dropdown-item\\" onClick={[Function: handleClick]} onKeyDown={[Function: handleKeyDown]}>
                                                                          <span>
                                                                            Help Articles
                                                                          </span>
                                                                        </a>
                                                                      </SafeAnchor>
                                                                    </DropdownItem>
                                                                    <SupportDocumentLink course={[undefined]}>
                                                                      <DropdownItem className=\\"support-document-link\\" target=\\"_blank\\" href=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-prodcms/media/documents/openstax_tutor_getting_started_guide_student.pdf\\" as={{...}} disabled={false}>
                                                                        <SafeAnchor target=\\"_blank\\" href=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-prodcms/media/documents/openstax_tutor_getting_started_guide_student.pdf\\" disabled={false} className=\\"support-document-link dropdown-item\\" onClick={[Function]}>
                                                                          <a target=\\"_blank\\" href=\\"https://d3bxy9euw4e147.cloudfront.net/oscms-prodcms/media/documents/openstax_tutor_getting_started_guide_student.pdf\\" className=\\"support-document-link dropdown-item\\" onClick={[Function: handleClick]} onKeyDown={[Function: handleKeyDown]}>
                                                                            <inject(TourAnchor) id=\\"menu-support-document\\">
                                                                              <Observer>
                                                                                <TourAnchor id=\\"menu-support-document\\" tourContext={{...}} tag=\\"div\\" className=\\"\\">
                                                                                  <div className=\\"tour-anchor\\" data-tour-anchor-id=\\"menu-support-document\\">
                                                                                    Getting Started Guide
                                                                                  </div>
                                                                                </TourAnchor>
                                                                              </Observer>
                                                                            </inject(TourAnchor)>
                                                                          </a>
                                                                        </SafeAnchor>
                                                                      </DropdownItem>
                                                                    </SupportDocumentLink>
                                                                    <BestPracticesGuide course={[undefined]} />
                                                                    <DropdownItem className=\\"-help-link\\" onSelect={[Function: res]} href=\\"/accessibility-statement/\\" onClick={[Function: res]} as={{...}} disabled={false}>
                                                                      <SafeAnchor href=\\"/accessibility-statement/\\" disabled={false} className=\\"-help-link dropdown-item\\" onClick={[Function]}>
                                                                        <a href=\\"/accessibility-statement/\\" className=\\"-help-link dropdown-item\\" onClick={[Function: handleClick]} onKeyDown={[Function: handleKeyDown]}>
                                                                          <span>
                                                                            Accessibility Statement
                                                                          </span>
                                                                        </a>
                                                                      </SafeAnchor>
                                                                    </DropdownItem>
                                                                  </div>
                                                                </DropdownMenu>
                                                              </div>
                                                            </ReactOverlaysDropdown>
                                                          </Dropdown>
                                                        </Responsive>
                                                      </SupportMenu>
                                                    </Observer>
                                                  </inject(SupportMenu)>
                                                </withRouter(inject(SupportMenu))>
                                                <withRouter(inject(PreviewAddCourseBtn)) course={[undefined]}>
                                                  <inject(PreviewAddCourseBtn) course={[undefined]} history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                                                    <Observer>
                                                      <PreviewAddCourseBtn course={[undefined]} history={{...}} location={{...}} match={{...}} staticContext={[undefined]} tourContext={{...}} />
                                                    </Observer>
                                                  </inject(PreviewAddCourseBtn)>
                                                </withRouter(inject(PreviewAddCourseBtn))>
                                                <UserMenu>
                                                  <Responsive desktop={{...}} mobile={{...}}>
                                                    <Dropdown className=\\"user-menu\\" navbar={false}>
                                                      <ReactOverlaysDropdown drop={[undefined]} show={[undefined]} alignEnd={[undefined]} onToggle={[Function]} focusFirstItemOnShow={[undefined]} itemSelector=\\".dropdown-item:not(.disabled):not(:disabled)\\">
                                                        <div onKeyDown={[Function: handleKeyDown]} className=\\"user-menu dropdown\\">
                                                          <DropdownToggle id=\\"user-menu\\" variant=\\"link\\" aria-label=\\"Account settings\\">
                                                            <Button onClick={[Function]} className=\\"dropdown-toggle\\" aria-haspopup={true} aria-expanded={false} id=\\"user-menu\\" variant=\\"link\\" aria-label=\\"Account settings\\" active={false} disabled={false} type=\\"button\\">
                                                              <button onClick={[Function]} aria-haspopup={true} aria-expanded={false} id=\\"user-menu\\" aria-label=\\"Account settings\\" disabled={false} type=\\"button\\" className=\\"dropdown-toggle btn btn-link\\">
                                                                <span className=\\"initials\\" />
                                                              </button>
                                                            </Button>
                                                          </DropdownToggle>
                                                          <DropdownMenu alignRight={true} flip={true} />
                                                        </div>
                                                      </ReactOverlaysDropdown>
                                                    </Dropdown>
                                                  </Responsive>
                                                </UserMenu>
                                              </Memo(wrappedComponent)>
                                            </Responsive>
                                          </Memo(wrappedComponent)>
                                        </Menus>
                                      </Component>
                                    </div>
                                  </nav>
                                </styled.nav>
                              </Navbar>
                              <withRouter(inject-with-courseContext(ServerErrorMonitoring))>
                                <inject-with-courseContext(ServerErrorMonitoring) history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                                  <ServerErrorMonitoring history={{...}} location={{...}} match={{...}} staticContext={[undefined]} courseContext={{...}} />
                                </inject-with-courseContext(ServerErrorMonitoring)>
                              </withRouter(inject-with-courseContext(ServerErrorMonitoring))>
                              <inject-with-modalManager(TermsModal)>
                                <TermsModal modalManager={{...}} />
                              </inject-with-modalManager(TermsModal)>
                              <Toasts toasts={{...}} />
                              <inject(CourseNagModal) course={[undefined]}>
                                <Observer>
                                  <CourseNagModal course={[undefined]} modalManager={{...}} tourContext={{...}} spyMode={{...}} />
                                </Observer>
                              </inject(CourseNagModal)>
                              <styled.div hasFooter={false} hasNavbar={true}>
                                <div className=\\"sc-fznZeY afmAJ\\">
                                  <Memo(wrappedComponent) app={[undefined]} />
                                  <p>
                                    hi i am body
                                  </p>
                                </div>
                              </styled.div>
                              <Navbar area=\\"footer\\" context={{...}} />
                            </div>
                          </styled.div>
                        </MobXProvider>
                      </TutorLayout>
                      <SpyModeContent>
                        <div className=\\"visible-when-debugging\\">
                          <div className=\\"tour-spy-info\\">
                            available regions: []; region tour ids: []; audience tags: []; tour tags: []; eligible tours: []; TOUR RIDE: &lt;none&gt;
                          </div>
                        </div>
                      </SpyModeContent>
                    </div>
                  </MobXProvider>
                </TourConductor>
              </inject-with-modalManager-spyMode(TourConductor)>
            </MobXProvider>
          </ModalManager>
          <a href=\\"#spy\\" tabIndex={-1} onClick={[Function: res]} className=\\"debug-toggle-link\\" aria-hidden=\\"true\\">
            π
          </a>
        </div>
      </MobXProvider>
    </SpyModeWrapper>
  </Router>
</MemoryRouter>"
`;
